### DISCUSSION TITLE
Why does AWS show a long json file with no human readable output flag?

### DISCUSSION THREAD

**Discussion by TheKevinWang** [aws/aws-cli#8813] [General]
So I tried to list VMs with `aws ec2 describe-instances` and I get this massive json file. In azure powershell, I would get a human readable table, but with the same level of fidelity as the aws json file, because they are objects. Why isn't there a --table or --human-readable flag for everything aws (I'm sure this isn't the only issue)? Seems like the azure powershell method is simply superior. Maybe they should release aws powershell. 

Guide suggests using jq query with `--query`. What a terrible idea in terms of usability lol. Imagine remembering and typing all this out all this just to do something as simple as list VMs. 
https://docs.aws.amazon.com/cli/latest/userguide/cli-services-ec2-instances.html

EDIT: turns out there is aws powershell, but it sucks
```
PS > Get-EC2Instance -ProfileName '<>' -debug 
DEBUG: Credentials obtained from stored profile named '<>'
DEBUG: Region obtained from stored region with value '<>'
VERBOSE: Invoking Amazon Elastic Compute Cloud (EC2) operation 'DescribeInstances' in region '<>'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): Y
Get-EC2Instance: The response ended prematurely.
```

**Comment by uldyssian-sh** [comment]
AWS CLI is designed to return machine-readable data by default (JSON), because it is a generic cross-platform CLI that targets automation first. The “human readable” view is done via output formatters plus filtering, rather than a single universal “pretty view” for every service and every shape.

You already have a built-in table/text output for essentially all AWS CLI commands:

aws ec2 describe-instances --output table

aws ec2 describe-instances --output text


What makes it look “massive” is that `describe-instances` returns a deeply nested structure (Reservations -> Instances). For a VM-style list, the intended usage is to select only the fields you want with `--query` and then render as a table.

Example: list instance id, name tag, state, type, AZ, private IP:

aws ec2 describe-instances \
  --query "Reservations[].Instances[].{InstanceId:InstanceId,Name:Tags[?Key=='Name']|[0].Value,State:State.Name,Type:InstanceType,AZ:Placement.AvailabilityZone,PrivateIP:PrivateIpAddress}" \
  --output table


If you want a simpler “VM list” that is easy to remember, keep it minimal:

aws ec2 describe-instances \
  --query "Reservations[].Instances[].{InstanceId:InstanceId,State:State.Name,Type:InstanceType}" \
  --output table


Why there isn’t one universal “--human-readable for everything”:

Different AWS services return very different data shapes (often nested, large, and service-specific). A single default “table” layout would either hide important fields, be inconsistent, or be extremely wide/noisy. That’s why AWS CLI provides:
- --output (json / yaml / text / table)
- --query (JMESPath) to pick the fields you actually want


About AWS PowerShell:

AWS Tools for PowerShell exists, but it’s PowerShell-native and follows PowerShell behaviors (like prompting). If you see “Confirm Continue with this operation?”, that’s PowerShell’s confirmation behavior. You can disable prompts for your session with:

$ConfirmPreference = 'None'

Or run the cmdlet with confirmation disabled (where supported):

Get-EC2Instance -Confirm:$false


In short:
- Use --output table/text for human-readable output
- Use --query to make it VM-like (InstanceId/Name/State/etc.)
- AWS CLI defaults to JSON because it is primarily an automation-first interface

**Reply by uldyssian-sh** [reply]
Thanks! If this helps, please consider marking it as “Helpful” or “Accepted” 😊
